packages/bruno-electron/src/ipc/network/index.js

```diff
@@ -124,31 +124,34 @@ const configureRequest = async (
 
   // client certificate config
   const clientCertConfig = get(brunoConfig, 'clientCertificates.certs', []);
-  for (let clientCert of clientCertConfig) {
-    const domain = interpolateString(clientCert.domain, interpolationOptions);
-
-    let certFilePath = interpolateString(clientCert.certFilePath, interpolationOptions);
-    certFilePath = path.isAbsolute(certFilePath) ? certFilePath : path.join(collectionPath, certFilePath);
 
-    let keyFilePath = interpolateString(clientCert.keyFilePath, interpolationOptions);
-    keyFilePath = path.isAbsolute(keyFilePath) ? keyFilePath : path.join(collectionPath, keyFilePath);
-
-    if (domain && certFilePath && keyFilePath) {
+  for (let clientCert of clientCertConfig) {
+    const domain = interpolateString(clientCert?.domain, interpolationOptions);
+    const type = clientCert?.type || 'cert';
+    if (domain) {
       const hostRegex = '^https:\\/\\/' + domain.replaceAll('.', '\\.').replaceAll('*', '.*');
-
       if (request.url.match(hostRegex)) {
-        try {
-          httpsAgentRequestFields['cert'] = fs.readFileSync(certFilePath);
-        } catch (err) {
-          console.log('Error reading cert file', err);
-        }
-
-        try {
-          httpsAgentRequestFields['key'] = fs.readFileSync(keyFilePath);
-        } catch (err) {
-          console.log('Error reading key file', err);
+        if (type === 'cert') {
+          try {
+            let certFilePath = interpolateString(clientCert?.certFilePath, interpolationOptions);
+            certFilePath = path.isAbsolute(certFilePath) ? certFilePath : path.join(collectionPath, certFilePath);
+            let keyFilePath = interpolateString(clientCert?.keyFilePath, interpolationOptions);
+            keyFilePath = path.isAbsolute(keyFilePath) ? keyFilePath : path.join(collectionPath, keyFilePath);
+
+            httpsAgentRequestFields['cert'] = fs.readFileSync(certFilePath);
+            httpsAgentRequestFields['key'] = fs.readFileSync(keyFilePath);
+          } catch (err) {
+            console.error('Error reading cert/key file', err);
```

Comment:
This errors should be returned to the user somehow (in a timeline like other types of erros?). In case of missing cert file I simply get an Error: socket hang up, as if no cert was configured at all, which is different than handling invalid cert files, or incorrect passphrase.